doc/Testing.md

```diff
@@ -98,6 +126,60 @@ For more documentation on testing, read the references listed at the bottom.
     }
     ```
 
+## Functionnal/Integration Testing
+
+- We use [integration_test](https://docs.flutter.dev/testing/integration-tests) to create functionnal/integration tests. You can create a test like this :
+
+	```cs
+	// Import the test package
+	import 'package:flutter_test/flutter_test.dart';
+	import 'package:integration_test/integration_test.dart';
+
+	void main() {
+	testWidgets('Get Dad Jokes', (WidgetTester tester) async {
+		// Arrange
+		await tester.pumpWidget(const App());
+
+		// Without this the dadjokesContainer isn't there yet.
+		await tester.pumpAndSettle();
+
+		var dadjokesContainer = find.byKey(const Key('DadJokesContainer'));
+
+		// Act
+		var dadJokes = find.descendant(
+		of: dadjokesContainer,
+		matching: find.byType(DadJokeListItem),
+		);
+
+		// Assert
+		expect(dadJokes, findsAtLeast(1));
+	});
+	}
+	```
+
+  To test different behaviors and interactions between the components of the app you need to simulate user interactions with the tester.tap(target) method like this:
+  ```cs
+  await tester.tap(dadJokes.first);
+  ```
+
+- To assert the result of a test, use `expect()` exactly like with unit tests.
+
+- tester.pumpAndSettle() is used both to trigger a frame change and to wait for the last pump to have settled before moving on. For example, we use it after pumping the app widget and we also use it when we naviguated and we want to update the UI.
+
+### Mocking
+
+For functionnal testing we use the decorator pattern because we are testing the actual behavior of the app so we don't wanna use mockito to mock the logic of the classes of the app. To use the decorator pattern you simply call the mocking repository and the method called setMocking(true). It's important to set the mocking in the main function before the tests because otherwise the mocking doesn't take effect. In most cases, except for api tests, the data sources(repositories) should be mocked for integration testing. 
+
+### Running the tests
+
+- You can run the integration test with this command :
+```bash
+flutter test integration_test --dart-define ENV=Development
+```
+  We need to set the env because the environment manager doesn't set one by default when we load the app it is set for each build accordingly so we need to manually set it. 
+
+- When testing locally it is important that the device on which you are testing is opened and running because the tests will interact with the app in real time. Otherwise it will throw unclear exceptions that don't point you at this issue.
```

Comment:
I personally didn't like how the first sentence was worded. Therefore, I've regenerated that first sentence with our friends ChatGPT & Grammarly `When testing locally, the device on which you are testing must be open and running, as the tests will interact with the app in real-time.`